lib/cloud_controller/organization_quota_usage_calculator.rb

```diff
@@ -0,0 +1,42 @@
+module VCAP::CloudController
+  class OrganizationQuotaUsageCalculator
+    def self.get_current_routes(org)
+      VCAP::CloudController::Route.dataset.join(:spaces, id: :space_id).
+        where(spaces__organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_service_instances(org)
+      VCAP::CloudController::ServiceInstance.dataset.join(:spaces, id: :space_id).
+        where(spaces__organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_domains(org)
+      VCAP::CloudController::Domain.dataset.
+        where(owning_organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_service_keys(org)
+      VCAP::CloudController::ServiceKey.dataset.join(:service_instances, id: :service_instance_id).
+        join(:spaces, id: :space_id).
+        where(spaces__organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_ports(org)
+      VCAP::CloudController::Route.dataset.join(:spaces, id: :space_id).
+        where(spaces__organization_id: org.id).
+        count || 0
+    end
+
+    def self.get_current_per_app_tasks(org)
+      VCAP::CloudController::TaskModel.dataset.where(state: TaskModel::RUNNING_STATE).
```

Comment:
You can reuse `org.running_and_pending_tasks_count` as it is done in `MaxAppTasksPolicy`.

`running_and_pending_tasks_count` uses nested selects, maybe you could change the query there to your JOIN thingy (and add `PENDING_STATE`).

Reply:
With the joins the statement looks now like this:
```
SELECT count(*) AS \"count\" FROM \"tasks\" INNER JOIN \"apps\" ON (\"apps\".\"guid\" = \"tasks\".\"app_guid\") INNER JOIN \"spaces\" ON (\"spaces\".\"guid\" = \"apps\".\"space_guid\") WHERE ((\"state\" IN ('PENDING', 'RUNNING')) AND (\"spaces\".\"organization_id\" = 13)) LIMIT 1
```
before there were 3 statements:
```
SELECT * FROM \"spaces\" WHERE (\"spaces\".\"organization_id\" = 2);
SELECT * FROM \"apps\" WHERE (\"apps\".\"space_guid\" IN ('1-2-3', '2-3-4'));
SELECT count(*) AS \"count\" FROM \"tasks\" WHERE ((\"tasks\".\"app_guid\" IN ('1-2-3', '2-3-4', '1-4-5', '3-4-5')) AND (\"state\" IN ('PENDING', 'RUNNING'))) LIMIT 1;
```